setup.py

```diff
@@ -1,134 +1,14 @@
 #!/usr/bin/env python
-
 """The script for setting up sqlfluff."""
-
-
-import sys
-
-if sys.version_info[0] < 3:
-    raise Exception("SQLFluff does not support Python 2. Please upgrade to Python 3.")
-
 import configparser
-from os.path import dirname
-from os.path import join
-
-from setuptools import find_packages, setup
-
+from setuptools import setup
 
 # Get the global config info as currently stated
 # (we use the config file to avoid actually loading any python here)
 config = configparser.ConfigParser()
 config.read(["src/sqlfluff/config.ini"])
 version = config.get("sqlfluff", "version")
```

Comment:
Due to custom nature of the version info in sqlfluff we keep the version logic in setup.py.

A better practice for single source of version info is to set the version entry in setup.cfg and then reference in src/sqlfluff/__init__.py:
```python
import importlib.metadata
__version__ = importlib.metadata.version('sqlfluff')
```
however, I believe that config.ini is referenced in a couple locations so it makes sense to keep the version logic as is for now and I'll investigate the above approach in a subsequent PR.